src/python/WMCore/Services/MicroService/Unified/Common.py

```diff
@@ -0,0 +1,290 @@
+        carry_on = {}
+        while True:
+            t = '%s%d' % (base, itask)
+            itask += 1
+            if t in workflow:
+                task = workflow[t]
+                if 'InputDataset' in task:
+                    dataset = task['InputDataset']
+                    if 'BlockWhitelist' in task and task['BlockWhitelist']:
+                        nevts, _ = getEventsLumis(dataset, task['BlockWhitelist'], eventsLumis)
+                    else:
+                        nevts, _ = getEventsLumis(dataset, eventsLumis=eventsLumis)
+                elif 'Input%s' % base in task:
+                    nevts = carry_on[task['Input%s' % base]]
+                elif 'RequestNumEvents' in task:
+                    nevts = float(task['RequestNumEvents'])
+                else:
+                    print("this is not supported, making it zero cput")
+                    nevts = 0
+                tpe = task.get('TimePerEvent', 1)
+                carry_on[task['%sName' % base]] = nevts
+                if 'FilterEfficiency' in task:
+                    carry_on[task['%sName' % base]] *= task['FilterEfficiency']
+                cput += tpe * nevts
+            else:
+                break
+    else:
+        nevts = float(workflow.get('RequestNumEvents', 0))
+        feff = float(workflow.get('FilterEfficiency', 1))
+        tpe = workflow.get('TimePerEvent', 1)
+        cput = nevts/feff * tpe
+
+    if cput is None:
+        return 0
+
+    if unit == 'm':
+        cput = cput / (60.)
+    if unit == 'h':
+        cput = cput / (60.*60.)
+    if unit == 'd':
+        cput = cput / (60.*60.*24.)
+    return cput
+
+def sigmoid(x):
+    "Sigmoid function"
+    return 1./(1 + math.exp(-x))
+
+def getNCopies(cpuHours, minN=2, maxN=3, weight=50000, constant=100000):
+    "Calculate number of copies for given workflow"
+    func = sigmoid(-constant/weight)
+    fact = (maxN - minN) / (1-func)
+    base = (func*maxN - minN)/(func-1)
+    return int(base + fact * sigmoid((cpuHours - constant)/weight))
+
+def teraBytes(size):
+    "Return size in TB"
+    return float(size)/float(1024**4)
+
+def ckey():
+    "Return user CA key either from proxy or userkey.pem"
+    return os.environ.get('X509_USER_PROXY', \
+        os.path.join(os.environ['HOME'], '.globus/userkey.pem'))
+
+def cert():
+    "Return user CA cert either from proxy or usercert.pem"
+    return os.environ.get('X509_USER_PROXY', \
+        os.path.join(os.environ['HOME'], '.globus/usercert.pem'))
+
+def stucktransferUrl():
+    "Return stucktransfer url"
+    return 'https://cms-stucktransfers.web.cern.ch/cms-stucktransfers'
+
+def dashboardUrl():
+    "Return dashboard url"
+    return 'http://dashb-ssb.cern.ch/dashboard/request.py'
+
+def monitoringUrl():
+    "Return monitoring url"
+    return 'http://cmsmonitoring.web.cern.ch/cmsmonitoring'
+
+def dbsUrl():
+    "Return DBS URL"
+    return 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader'
+
+def reqmgrUrl():
```

Comment:
Not really a review, just skimming the code. 
Valentin, can you set ALL these variables in a configuration file instead? Having a function to simply return a hardcoded value doesn't look right to me.